index.js

```diff
@@ -0,0 +1,210 @@
+    },
+    "None": {
+        "None": ""
+    }
+}
+
+const EMAIL_REGEX = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+
+const ALL_LICENSES_BY_KEYS = {};
+
+const CHOICES = generateChoices();
+
+// function that generates choices array out of the licenses object hard coded above as well as the object from which the badge is fetched
+
+function generateChoices() {
+    var choicesArr = []
+    Object.keys(LICENSES).forEach(type => {
+        choicesArr.push(new inquirer.Separator("----" + type + "----"));
+        var licensesByType = LICENSES[type];
+        Object.assign(ALL_LICENSES_BY_KEYS, licensesByType);
+        Object.keys(licensesByType).forEach(license => {
+            choicesArr.push(license);
+        })
+    })
+    return choicesArr;
+}
+
+// Questions list
+
+const QUESTIONS = [
+    {
+        type: "input",
+        name: "gitUsername",
+        message: "What is your Github Username?"
+    },
+    {
+        type: "input",
+        name: "email",
+        message: "What is your email?",
+        validate: function (input) {
+            if (EMAIL_REGEX.test(input)) return true;
+            return "Please enter a valid email!"
+        }
+    },
+    {
+        type: "input",
+        name: "title",
+        message: "What is the title of the Readme?"
+    },
+    {
+        type: "list",
+        name: "license",
+        message: "Please choose a license:",
+        choices: CHOICES
+    },
+    {
+        type: "input",
+        name: "description",
+        message: "Enter the Readme description:"
+    },
+    {
+        type: "input",
+        name: "usage_information",
+        message: "Please provide usage information"
+    },
+    {
+        type: "input",
+        name: "installation",
+        message: "Please provide installation steps, comma separated:"
+    },
+    {
+        type: "input",
+        name: "contributing",
+        message: "Please provide contibution instructions (If left blank, a default response will be provided):",
+        default: "Please follow [Contributor Covenant](https://www.contributor-covenant.org/), which is the is an industry standard."
+    },
+    {
+        type: "input",
+        name: "testing",
+        message: "Please provide testing methods:",
+    }
+]
+
+
+// Main Function
+
+inquirer
+    .prompt(QUESTIONS)
+    .then((responses) => {
+        var readMe = new README(responses);
+        writeReadmeFile(readMe);
+    });
+
+
+// README Constructor
+const README = function (responses) {
+    this.fileName = "README.md";
+    this.title = `#  ${responses.title}\n\n`;
+    var licenseBadge = ALL_LICENSES_BY_KEYS[responses.license] + `\n\n`;
+    var contributionBadge = responses.contributing.includes("https://www.contributor-covenant.org/") ? "[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg)](code_of_conduct.md)" : "";
+    this.badges = licenseBadge + " " + contributionBadge + '\n\n';
```

Comment:
it may be easier to template string all of this logic directly for better readability and less code lines